parsely/src/main/java/com/parsely/parselyandroid/SendEvents.kt

```diff
@@ -0,0 +1,57 @@
+package com.parsely.parselyandroid
+
+import com.parsely.parselyandroid.JsonSerializer.toParselyEventsPayload
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+internal class SendEvents(
+    private val flushManager: FlushManager,
+    private val localStorageRepository: LocalStorageRepository,
+    private val parselyAPIConnection: ParselyAPIConnection,
+    private val scope: CoroutineScope
+) {
+
+    private val mutex = Mutex()
+
+    operator fun invoke(isDebug: Boolean) {
```

Comment:
How do you feel about renaming `isDebug` to `skipSendingEvents` or something similar?

`isDebug` could mean anything and when such an argument is available, sometimes it's used for several different things within the implementation. Having a specific name should help with self-documentation and it'll discourage developers from overloading the argument with multiple responsibilities.

In this context, I think it's kind of clear what `isDebug` is used for - but I think a specific name will be more future proof.

Reply:
I agree with your arguments: +1 from me for renaiming `isDebug` to `skipSendingEvents`. Additionally, I renamed `SendEvents` to `FlushQueue` because I believe the possibility to run "send events with skipping sending events" might be confusing. Addressed in c9872d4.